marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-step-detail/mapping-step-detail.tsx

```diff
@@ -253,14 +255,13 @@ const MappingStepDetail: React.FC = () => {
           }
         }
         let nestedDoc: any = [];
-        let docRoot = parsedDoc["envelope"] ? parsedDoc["envelope"]["instance"] : parsedDoc;
+        let docRoot = mappingStep.sourceRecordScope === "entireRecord" ?  parsedDoc : parsedDoc["envelope"]["instance"] ;
```

Comment:
Is there a way for the backend to handle this? We don't want the UI assuming knowledge of what a document should look like. The backend just return the portion of the document that should be displayed in the "Source Doc" view based on how the step is configured.